src/core/initial_conditions.jl

```diff
@@ -1,383 +1,128 @@
-    _make_initial_conditions!(
-        psi_container,
-        devices,
-        ICKey(DeviceStatus, T),
-        _make_initial_condition_active_power,
-        _get_active_power_status_value,
-        # Doesn't require Cache
-    )
-end
-
-function output_init(
-    psi_container::PSIContainer,
-    devices::IS.FlattenIteratorWrapper{T},
-) where {T <: PSY.HydroGen}
-    _make_initial_conditions!(
-        psi_container,
-        devices,
-        ICKey(DevicePower, T),
-        _make_initial_condition_active_power,
-        _get_active_power_output_value,
-        # Doesn't require Cache
-    )
-
-    return
-end
-
-function duration_init(
-    psi_container::PSIContainer,
-    devices::IS.FlattenIteratorWrapper{T},
-) where {T <: PSY.HydroGen}
-    for key in (ICKey(TimeDurationON, T), ICKey(TimeDurationOFF, T))
-        _make_initial_conditions!(
-            psi_container,
-            devices,
-            key,
-            _make_initial_condition_active_power,
-            _get_active_power_duration_value,
-            TimeStatusChange,
-        )
-    end
-
-    return
-end
-
-# TODO: This IC needs a cache for Simulation over long periods of time
-function storage_energy_init(
-    psi_container::PSIContainer,
-    devices::IS.FlattenIteratorWrapper{T},
-) where {T <: PSY.HydroGen}
-    key = ICKey(EnergyLevel, T)
-    _make_initial_conditions!(
-        psi_container,
-        devices,
-        key,
-        _make_initial_condition_reservoir_energy,
-        _get_reservoir_energy_value,
-        StoredEnergy,
-    )
-
-    return
-end
-
-function _make_initial_conditions!(
-    psi_container::PSIContainer,
-    devices::IS.FlattenIteratorWrapper{T},
-    key::ICKey,
-    make_ic_func::Function,
-    get_val_func::Function,
-    cache = nothing,
-) where {T <: PSY.Device}
-    length_devices = length(devices)
-
-    if !has_initial_conditions(psi_container, key)
-        @debug "Setting $(key.ic_type) initial conditions for the status of all devices $(T) based on system data"
-        ini_conds = Vector{InitialCondition}(undef, length_devices)
-        set_initial_conditions!(psi_container, key, ini_conds)
-        for (ix, dev) in enumerate(devices)
-            ini_conds[ix] = make_ic_func(psi_container, dev, get_val_func(dev, key), cache)
-        end
-    else
-        ini_conds = get_initial_conditions(psi_container, key)
-        ic_devices = Set((IS.get_uuid(ic.device) for ic in ini_conds))
-        for dev in devices
-            IS.get_uuid(dev) in ic_devices && continue
-            @debug "Setting $(key.ic_type) initial conditions for the status device $(PSY.get_name(dev)) based on system data"
-            push!(
-                ini_conds,
-                make_ic_func(psi_container, dev, get_val_func(dev, key), cache),
-            )
-        end
-    end
-
-    @assert length(ini_conds) == length_devices
-    return
-end
-
-function _make_initial_condition_active_power(
-    psi_container,
+"""
```

Comment:
@jd-lara These next two functions are a starting point for discussing the user interface for creating initial conditions.  This likely requires too much from the user.  What would you like to simplify?

Reply:
We should move this function to a different PR to discuss the interface. My first impression is: 

1. users should not need to pass an `InitialConditions` to add them. 
2. It should be a single entry point like a kwarg passed into the stage/operation problem